File: Daily Challenge/March 2025/Count Days Without Meetings.cpp
class Solution {
public:
  int countDays(int days, vector<vector<int>>& meetings) {
    sort(meetings.begin(), meetings.end());

    int daysWithoutMeetings = 0;
    int lastDay = 0;

    for (auto meeting : meetings) {
      if (meeting[0] > lastDay) {
        daysWithoutMeetings += meeting[0] - lastDay - 1;
      }
      lastDay = max(lastDay, meeting[1]);
    }
    daysWithoutMeetings += days - lastDay;

    return daysWithoutMeetings;
  }
};

File: Daily Challenge/March 2025/Count Days Without Meetings.cs
public class Solution {
  public int CountDays(int days, int[][] meetings) {
    
    meetings = meetings.OrderBy(x => x[0]).ToArray();

    int daysWithoutMeetings = 0;
    int lastMeeting = 0;

    foreach (var meeting in meetings) {
      if (meeting[0] > lastMeeting) {
        daysWithoutMeetings += meeting[0] - lastMeeting - 1;
      }

      lastMeeting = Math.Max(lastMeeting, meeting[1]);
    }
    daysWithoutMeetings += days - lastMeeting;

    return daysWithoutMeetings;
  }
}